File: src/skill-catalog.js
export const SKILL_CATALOG = Object.freeze([
  { id: 'S01', name: 'Arc Push', tag: 'Knockback', castType: 'Direction', cooldownSec: 5, pushForce: 1.0, price: 2 },
  { id: 'S02', name: 'Blast Palm', tag: 'Knockback', castType: 'Target', cooldownSec: 7, pushForce: 1.3, price: 3 },
  { id: 'S03', name: 'Shock Ring', tag: 'Knockback', castType: 'Area', cooldownSec: 10, pushForce: 1.1, price: 4 },
  { id: 'S04', name: 'Overdrive Wave', tag: 'Knockback', castType: 'Direction', cooldownSec: 14, pushForce: 1.6, price: 6 },
  { id: 'S05', name: 'Ember Bolt', tag: 'Damage', castType: 'Direction', cooldownSec: 4, pushForce: 0.2, price: 2 },
  { id: 'S06', name: 'Void Spear', tag: 'Damage', castType: 'Target', cooldownSec: 8, pushForce: 0.1, price: 4 },
  { id: 'S07', name: 'Blink Step', tag: 'Mobility', castType: 'Self', cooldownSec: 10, pushForce: 0.0, price: 3 },
  { id: 'S08', name: 'Dash Burst', tag: 'Mobility', castType: 'Direction', cooldownSec: 12, pushForce: 0.4, price: 4 },
  { id: 'S09', name: 'Slow Field', tag: 'Control', castType: 'Area', cooldownSec: 12, pushForce: 0.0, price: 3 },
  { id: 'S10', name: 'Silence Pulse', tag: 'Control', castType: 'Area', cooldownSec: 15, pushForce: 0.0, price: 5 },
  { id: 'S11', name: 'Guard Shell', tag: 'Defense', castType: 'Self', cooldownSec: 13, pushForce: 0.0, price: 4 },
  { id: 'S12', name: 'Barrier Wall', tag: 'Defense', castType: 'Area', cooldownSec: 16, pushForce: 0.0, price: 5 },
]);

export const SKILL_BY_ID = Object.freeze(Object.fromEntries(SKILL_CATALOG.map((s) => [s.id, s])));

File: Assets/Warlock/Scripts/Mvp/SkillCatalog.cs
using System.Collections.Generic;
using System.Linq;

namespace Warlock.Mvp
{
    public readonly struct SkillCard
    {
        public SkillCard(string id, string name, string tag, string castType, int cooldownSec, float pushForce, int price)
        {
            Id = id;
            Name = name;
            Tag = tag;
            CastType = castType;
            CooldownSec = cooldownSec;
            PushForce = pushForce;
            Price = price;
        }

        public string Id { get; }
        public string Name { get; }
        public string Tag { get; }
        public string CastType { get; }
        public int CooldownSec { get; }
        public float PushForce { get; }
        public int Price { get; }
    }

    public static class SkillCatalog
    {
        public static readonly IReadOnlyList<SkillCard> Fixed12 = new[]
        {
            new SkillCard("S01", "Arc Push", "Knockback", "Direction", 5, 1.0f, 2),
            new SkillCard("S02", "Blast Palm", "Knockback", "Target", 7, 1.3f, 3),
            new SkillCard("S03", "Shock Ring", "Knockback", "Area", 10, 1.1f, 4),
            new SkillCard("S04", "Overdrive Wave", "Knockback", "Direction", 14, 1.6f, 6),
            new SkillCard("S05", "Ember Bolt", "Damage", "Direction", 4, 0.2f, 2),
            new SkillCard("S06", "Void Spear", "Damage", "Target", 8, 0.1f, 4),
            new SkillCard("S07", "Blink Step", "Mobility", "Self", 10, 0.0f, 3),
            new SkillCard("S08", "Dash Burst", "Mobility", "Direction", 12, 0.4f, 4),
            new SkillCard("S09", "Slow Field", "Control", "Area", 12, 0.0f, 3),
            new SkillCard("S10", "Silence Pulse", "Control", "Area", 15, 0.0f, 5),
            new SkillCard("S11", "Guard Shell", "Defense", "Self", 13, 0.0f, 4),
            new SkillCard("S12", "Barrier Wall", "Defense", "Area", 16, 0.0f, 5)
        };

        public static readonly IReadOnlyDictionary<string, SkillCard> ById = Fixed12.ToDictionary(s => s.Id, s => s);
    }
}
